fonduer/parser/parser.py

```diff
@@ -269,13 +283,117 @@ def _parse_figure_node(self, node, state):
 
         return state
 
-    def _parse_sentence(self, node, state):
+    def _parse_sentence(self, paragraph, node, text, field, state):
         """Parse the Sentences of the node.
 
         :param node: The lxml node to parse
         :param state: The global state necessary to place the node in context
             of the document as a whole.
         """
+        # Lingual Parse
+        document = state["document"]
+        for parts in self.lingual_parse(document, text):
+            (_, _, _, char_end) = split_stable_id(parts["stable_id"])
+            parts["document"] = document
+            parts["sentence_num"] = state["sentence"]["idx"]
+            abs_sentence_offset_end = (
+                state["sentence"]["abs_offset"]
+                + parts["char_offsets"][-1]
+                + len(parts["words"][-1])
+            )
+            parts["stable_id"] = construct_stable_id(
+                document,
+                "sentence",
+                state["sentence"]["abs_offset"],
+                abs_sentence_offset_end,
+            )
+            state["sentence"]["abs_offset"] = abs_sentence_offset_end
+            if self.structural:
+                context_node = node.getparent() if field == "tail" else node
```

Comment:
I know this is old code. Just make sure the logic here is correct. Will the the tail node be the sibling of the text node or now?